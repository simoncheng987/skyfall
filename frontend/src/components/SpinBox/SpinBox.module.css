input[type='number']::-webkit-inner-spin-button {
  opacity: 1;
  padding: 10px;
  margin: 0;
}

.spinbox {
  font-size: 32px;
  caret-color: transparent;
  text-align: center;
  border-radius: 15px;
  border-color: #1c1c1c;
  border-width: 3px;
  height: 85px;
  width: 100%;
}

.spinbox:hover {
  cursor: default;
}

.spinbox:focus {
  outline-width: unset;
}
