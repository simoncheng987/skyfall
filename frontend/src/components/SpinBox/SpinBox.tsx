import React from 'react';
import styles from './SpinBox.module.css';

interface SpinBoxProps {
  /**
   * Action to be performed when value has been changed.
   */
  onChange?: React.ChangeEventHandler<HTMLInputElement>;

  /**
   * Minimum value of the spinbox.
   */
  min?: number;

  /**
   * Maximum value of the spinbox.
   */
  max?: number;

  /**
   * Default value of the spinbox.
   */
  defaultValue?: number;

  /**
   * Additional class names to style this component from the outside.
   */
  className?: string;
}

/**
 * A styled spinbox component.
 */
export default function SpinBox({ onChange, min, max, defaultValue, className }: SpinBoxProps) {
  return (
    <input
      type="number"
      min={min}
      max={max}
      defaultValue={defaultValue}
      onKeyDown={(e) => e.preventDefault()}
      className={`${styles.spinbox} ${className}`}
      onChange={onChange}
    />
  );
}

SpinBox.defaultProps = {
  onChange: () => {},
  min: Math.min,
  max: Math.max,
  defaultValue: 0,
  className: '',
};
