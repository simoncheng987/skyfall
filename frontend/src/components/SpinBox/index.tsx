import SpinBox from './SpinBox';

export default SpinBox;
