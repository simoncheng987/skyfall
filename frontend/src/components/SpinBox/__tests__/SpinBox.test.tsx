import React from 'react';
import renderer from 'react-test-renderer';
import SpinBox from '../SpinBox';

describe('Dropdown', () => {
  const className = 'testClassName';

  it('snapshot matches with className props', () => {
    const tree = renderer
      .create(<SpinBox min={0} max={10} defaultValue={5} className={className} />)
      .toTree();
    expect(tree).toMatchSnapshot();
  });
});
