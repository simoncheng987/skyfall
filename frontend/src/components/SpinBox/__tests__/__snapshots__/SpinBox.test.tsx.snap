// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown snapshot matches with className props 1`] = `
Object {
  "instance": null,
  "nodeType": "component",
  "props": Object {
    "className": "testClassName",
    "defaultValue": 5,
    "max": 10,
    "min": 0,
    "onChange": [Function],
  },
  "rendered": Object {
    "instance": null,
    "nodeType": "host",
    "props": Object {
      "className": "spinbox testClassName",
      "defaultValue": 5,
      "max": 10,
      "min": 0,
      "onChange": [Function],
      "onKeyDown": [Function],
      "type": "number",
    },
    "rendered": Array [],
    "type": "input",
  },
  "type": [Function],
}
`;
